src/pages/workspaces/workspace/DeleteWorkspaceModal.js

```diff
@@ -1,23 +1,73 @@
 import _ from 'lodash/fp'
 import { useState } from 'react'
-import { div, h, label, span } from 'react-hyperscript-helpers'
+import { div, h, label, p, span } from 'react-hyperscript-helpers'
 import { ButtonPrimary, Link, spinnerOverlay } from 'src/components/common'
+import { warningBoxStyle } from 'src/components/data/data-utils'
+import { icon } from 'src/components/icons'
 import { TextInput } from 'src/components/input'
 import Modal from 'src/components/Modal'
 import { Ajax } from 'src/libs/ajax'
-import { bucketBrowserUrl } from 'src/libs/auth'
+import { bucketBrowserUrl, getUser } from 'src/libs/auth'
+import colors from 'src/libs/colors'
 import { reportError } from 'src/libs/error'
+import { isAppDeletable } from 'src/libs/runtime-utils'
 import * as Utils from 'src/libs/utils'
 
 
 const DeleteWorkspaceModal = ({ workspace: { workspace: { namespace, name, bucketName } }, onDismiss, onSuccess }) => {
   const [deleting, setDeleting] = useState(false)
   const [deleteConfirmation, setDeleteConfirmation] = useState('')
+  const [loadingApps, setLoadingApps] = useState(true)
```

Comment:
this is used to ensure the spinner stays present until the apps are loaded, so the user can't 'quickly delete' it before we detect apps that cant be deleted. It's not super relevant in the happy path where the call returns almost instantly, but if you modify your network speed to 3G or the like in the console and remove this its quite jarring

Reply:
This seems unnecessary; why not just check if `apps === undefined` to see if they've been loaded yet?